CONTRIBUTING.md

```diff
@@ -0,0 +1,177 @@
+# The Developer Workflow 
+
+-   [General Notes](#general-notes)
+-   [Issuing a Pull Request](#issuing-a-pull-request)
+-   [Reviewing a Pull Request](#reviewing-a-pull-request)
+-   [Running Tests](#running-tests)
+-   [Cautions](#cautions)
+-   [An Example](#an-example)
+    -   [Introduction](#introduction)
+    -   [Acquiring Moltres and Workflow](#acquiring-moltres-and-workflow)
+    -   [See also](#see-also)
+-   [Releases](#releases)
+
+------------------------------------------------------------------------
+
+## General Notes
+
+-   The terminology we use is based on the
+    [Integrator Workflow ](http://en.wikipedia.org/wiki/Integrator_workflow).
+
+-   Use a branching workflow similar to the one described
+    in the [progit gitbook](https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows).
+
+-   Keep your own "main" and "devel" branches in sync with the
+    main repository's "main" and "devel" branches. Specifically,
+    do not push your own commits directly to your "main" and
+    "devel" branches.
+
+-   Any commit should *pass all tests* (see [Running Tests](#running-tests)).
+
+-   See the [An Example](#an-example) section below for a full walk through.
+
+-   In addition to a review of the algorithmic and logical changes in
+    your contribution, it will be reviewed on a variety of levels
+    including such things as style, documentation, tests, etc. While
+    such review can appear tedious, these aspects are important for the
+    long term success of the project.
+
+## Issuing a Pull Request
+
+-   When you are ready to move changes from one of your topic branches
+    into the "devel" branch, it must be reviewed and accepted by
+    another develer.
+
+-   You may want to review this
+    [tutorial](https://help.github.com/articles/using-pull-requests/)
+    before you make a pull request to the devel branch.
+
+## Reviewing a Pull Request
+
+-   Look over the code.
+
+-   Make inline review comments concerning improvements.
+
+-   Wait for the Continuous Integration service to show full test
+    passage.
+
+-   Click the green "Merge Pull Request" button.
+
+-   Note: if the button is not available, the requester needs to merge
+    or rebase from the current HEAD of the blessed's "devel"
+    (or "master") branch.
+
+## Running Tests
+
+TBD
```

Comment:
This will need to be changed when I use this document as a template. I may consider adding something for this particular document on how scripts are tricky to test